components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -864,6 +864,64 @@ void BraveWalletService::MigrateUserAssetsAddIsNFT(PrefService* prefs) {
   prefs->SetBoolean(kBraveWalletUserAssetsAddIsNFTMigrated, true);
 }
 
+// static
+void BraveWalletService::MigrateUserAssetsAddFilecoinEvm(PrefService* prefs) {
+  if (prefs->GetBoolean(kBraveWalletFilecoinEvmNetworksAdded)) {
+    return;
+  }
+
+  {
+    const std::string filecoin_evm_mainnet =
+        GetKnownEthNetworkId(mojom::kFilecoinEtheriumMainnetChainId);
+    const std::string filecoin_evm_testnet =
+        GetKnownEthNetworkId(mojom::kFilecoinEtheriumTestnetChainId);
+
+    ScopedDictPrefUpdate update(prefs, kBraveWalletUserAssets);
+    auto& user_assets_pref = update.Get();
+    auto* dict = user_assets_pref.EnsureDict(kEthereumPrefKey);
+
+    // Filecoin mainnet
+    {
+      base::Value::List* list = dict->EnsureList(filecoin_evm_mainnet);
+      auto value = base::Value(GetEthNativeAssetFromChain(
+          GetKnownChain(nullptr, filecoin_evm_mainnet, mojom::CoinType::ETH)));
+      if (std::find_if(list->begin(), list->end(), [&value](auto& v) {
+            return value == v;
+          }) == list->end()) {
+        list->Insert(list->begin(), std::move(value));
+      }
+    }
+
+    // Filecoin testnet
+    {
+      base::Value::List* list = dict->EnsureList(filecoin_evm_testnet);
+      auto value = base::Value(GetEthNativeAssetFromChain(
+          GetKnownChain(nullptr, filecoin_evm_testnet, mojom::CoinType::ETH)));
+      if (std::find_if(list->begin(), list->end(), [&value](auto& v) {
+            return value == v;
+          }) == list->end()) {
+        list->Insert(list->begin(), std::move(value));
+      }
+    }
+  }
```

Comment:
Can we migrate this part using existing MigrateUserAssetsAddPreloadingNetworks?
See https://github.com/brave/brave-core/pull/14080/files where we added another network and just updating kBraveWalletUserAssetsAddPreloadingNetworksMigrated.